src/figuresClasses.ts

```diff
@@ -1,19 +1,80 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side must be greater than zero');
+    }
+
+    if ((a + b <= c) || (a + c <= b) || (b + c <= a)) {
+      throw new Error('The hypotenuse is shorter than the sum of the legs');
+    }
+  }
+
+  getArea(): number {
+    const halfOfPerimeter: number = (this.a + this.b + this.c) / 2;
 
+    const triangleArea = Math.sqrt(
+      halfOfPerimeter
+      * (halfOfPerimeter - this.a)
+      * (halfOfPerimeter - this.b)
+      * (halfOfPerimeter - this.c),
+    );
+
+    return Math.floor(triangleArea * 100) / 100;
```

Comment:
Optional: you round with the same logic twice over the code. You also could create a separate helper function - `rounder` for example for this purpose.